tempo_embeddings/embeddings/weaviate_database.py

```diff
@@ -0,0 +1,278 @@
+import json
+import logging
+import os
+import platform
+from typing import Any
+from typing import Iterable
+from typing import TypeVar
+import weaviate
+import weaviate.classes as wvc
+from transformers import AutoTokenizer
+from weaviate.classes.query import Filter
+from weaviate.classes.query import MetadataQuery
+from weaviate.util import generate_uuid5
+from ..text.corpus import Corpus
+from ..text.highlighting import Highlighting
+from ..text.passage import Passage
+from .vector_database import VectorDatabaseManagerWrapper
+
+
+Collection = TypeVar("Collection")
+logger = logging.getLogger(__name__)
+
+class WeaviateDatabaseManager(VectorDatabaseManagerWrapper):
+    """A Weaviate Database can create N collections and will always use THE SAME embedder function and tokenizer for all collections.
+    To create collections with different embedders one needs to create separate databases
+    """
+
+    def __init__(
+        self,
+        db_path: str = "weaviate_default_db",
+        embedder_name: str = None,
+        embedder_config: dict[str, Any] = None,
+        batch_size: int = 8,
+    ):
+        super().__init__(batch_size)
+        self.db_path = db_path
+        self.embedder_name = embedder_name
+        self.embedder_config = embedder_config or {"type": "default"}
+        self.tokenizer = AutoTokenizer.from_pretrained(embedder_name) if embedder_name else None
+        self.model = None
+        self.client = None
+        self.weaviate_headers = {}
+        
+        if self.embedder_config.get("type") == "hf":
+            self.embedding_function = wvc.config.Configure.Vectorizer.text2vec_huggingface(model=self.embedder_name)
+            self.weaviate_headers = {"X-HuggingFace-Api-Key": self.embedder_config["api_key"]}
+        elif self.embedder_config.get("type") == "custom_model" or self.embedder_config.get("type") == "default":
+            try:
+                self.model = self.embedder_config["model"]
+                self.model.batch_size = self.batch_size
+                self.embedding_function = wvc.config.Configure.Vectorizer.none()
+            except KeyError:
+                logger.error("If the type is 'custom_model' or 'default' you should pass the model object under Key 'model'")
```

Comment:
It is often useful to log the original error:
```suggestion
            except KeyError as e:
                logger.error(f"If the type is 'custom_model' or 'default' you should pass the model object under Key 'model': {e}")
```